fix: addApplicationDelegate unavailable in iOS

armandsLa opened the pull request:
I'm proposing a simple fix to avoid errors caused by `NS_EXTENSION_UNAVAILABLE_IOS` when adding `home_widget` to the widget target. This fix uses `performSelector` to bypass the `NS_EXTENSION_UNAVAILABLE_IOS` macro. It might not be ideal, but it works and is very simple.

ABausG commented:
Ooh! This looks very nice. I tried to work around this by using some `@available` annotations but those seem to not correctly check at runtime. Will have closer look later today!

ABausG commented on ios/Classes/SwiftHomeWidgetPlugin.swift:
```diff
@@ -30,7 +36,14 @@ public class SwiftHomeWidgetPlugin: NSObject, FlutterPlugin, FlutterStreamHandle
     eventChannel.setStreamHandler(instance)
 
-    registrar.addApplicationDelegate(instance)
+    guard isRunningInAppExtension() == false else {
+      return
+    }
```
I'm not a swift/iOS expert so could you maybe elaborate how this (and checking for the extension via the extension) is different to wrapping the function invocation with an `#available(OSApplicationExtension, unavailable) check?

armandsLa replied:
Unfortunately, `#available` cannot be used to detect the target type. The code added inside `#available(OSApplicationExtension unavailable)` will not run anywhere. Checking the path extension is the only way to detect that code is being executed from an extension, not the main app, at least as far as I know.

The only alternative I can think of would be to create a separate configuration method that executes `addApplicationDelegate` and require users to call it from `didFinishLaunchingWithOptions`. However, I think that this would be a worse solution to this problem.

ABausG reviewed (commented):
Thank you so much for this!

I'd love to hear some more reasoning about your approach in comparison to an available check. Mostly to learn a bit more about iOS!

Also could you possibly add  the pod spec adjustments needed for Interactivity to the example pod file? I noticed that this way the Integration Test Would fail without adjustments so it would be a great check

```ruby
target 'YourWidgetExtension' do
   use_frameworks!
   use_modular_headers!

   pod 'home_widget', :path => '.symlinks/plugins/home_widget/ios'
end
```

ABausG commented:
@armandsLa Do you still have time to finish this off?

armandsLa commented:
Sorry for the late response; I've been very busy lately.

I've made changes to the Podfile as requested. I moved every target to the root level and added the necessary configurations. Previously, the targets were nested, which I had never seen before. Usually, every target is at the root level, and optionally, a `shared_pods` method is defined to keep everything DRY. In this case, it was not necessary, though.

ABausG commented:
Had to make some adjustments to the Podfile in order to have integration tests pass. The nesting seems to be important to work correctly. Given that the `RunnerTest` nesting is also done by the default Flutter Project I feel this is fine. Most important thing the tests are passing again
